stdlib/public/core/NewtypeWrapper.swift

```diff
@@ -15,11 +15,16 @@
 /// attribute.
 public protocol _SwiftNewtypeWrapper : RawRepresentable { }
 
-extension _SwiftNewtypeWrapper where Self.RawValue : Hashable {
+extension _SwiftNewtypeWrapper where Self: Hashable, Self.RawValue : Hashable {
   @inlinable // FIXME(sil-serialize-all)
   public var hashValue: Int {
     return rawValue.hashValue
   }
+
+  @inlinable // FIXME(sil-serialize-all)
+  public func _hash(into hasher: inout _Hasher) {
+    hasher.append(rawValue)
```

Comment:
Nice catch. Is this something that users are likely to run into, or is it because of the special relationship between newtype'd imports and bridging?

Reply:
The crash was caused by a particularly weird interplay between `_hash(into:)`, newtype, `AnyHashable` *and* bridging.

Luckily, I don't see how it could apply to user code. For example, struct `Foo` below produces the same `hashValue` as its label, but it doesn't hash the same way through `_hash(into:)`. This isn't an issue, though, because the only way to put them in the same hash table is through AnyHashable, and since they don't have the same AnyHashable representation, they wouldn't compare equal even if they had matching hashes:

```swift
struct Foo: Hashable {
  var label: String
  var hashValue: Int { return label.hashValue }
  static func ==(left: Foo, right: Foo) -> Bool { return left.label == right.label }
}

Foo(label: "foo").hashValue == "foo".hashValue // ==> true
(Foo(label: "foo") as AnyHashable) == ("foo" as AnyHashable) // ==> false
let s: Set<AnyHashable> = [Foo(label: "foo"), "foo"]
s.count // ==> 2
```

The only visible effect seems to be that Dictionary/Set ordering is different for `Foo` and `String`, but hash table ordering is explicitly unstable, so code that depends on that is already broken.

Reply:
Okay. Imported newtypes are specifically weird because we need hashing consistency for the newtype, the type it wraps, and potentially the bridged type (`SomeNewType`/`String`/`NSString`), so yeah--unlikely to affect anyone else.